app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt

```diff
@@ -33,21 +34,37 @@ class AuthService(private val authApi: AuthApi,
     fun isLoggedIn() = authHolder.isLoggedIn()
 
     fun logout() {
-        userMap.remove(authHolder.getId())
-
+        val userFlowable = userDao.getUser(authHolder.getId())
+        userFlowable.map {
+            Single.fromCallable {
+                userDao.deleteUser(it)
+            }.subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({
+                        Timber.d("User logged out!")
+                    }
+                    ) {
+                        Timber.e(it, "Error logging out user!")
+                    }
+        }
         authHolder.token = null
     }
 
-    fun getProfile(id: Long = authHolder.getId()): Single<User> {
-        val user = userMap[id]
-        if (user != null)
-            return Single.just(user)
 
-        return authApi.getProfile(id)
-                .map {
-                    userMap[id] = it
-                    it
-                }
+    fun getProfile(id: Long = authHolder.getId()): Flowable<User> {
+        val userFlowable = userDao.getUser(id)
+        userFlowable.switchIfEmpty{
```

Comment:
This will never be empty

Reply:
what should be the condition to fetch user from network then? (when user not found from local db)

Reply:
When user is not present, an error is thrown, verify that this happens. If so, use onErrorResumeNext

Reply:
tried inside(getProfile())
```userFlowable.doOnError{
            Timber.e("Error fetching user")
        }
```
but no error was shown in log trace

Reply:
What happened when you subscribed it. RxJava guarantees that you'll either get data or error. If you didn't get error, you must have gotten data, and if there is no user with that ID, you couldn't have